activiti-api/activiti-process-runtime-api-impl/src/main/java/org/activiti/runtime/api/impl/ProcessAdminRuntimeImpl.java

```diff
@@ -0,0 +1,208 @@
+import org.activiti.runtime.api.NotFoundException;
+import org.activiti.runtime.api.ProcessAdminRuntime;
+import org.activiti.runtime.api.conf.ProcessRuntimeConfiguration;
+import org.activiti.runtime.api.model.ProcessDefinition;
+import org.activiti.runtime.api.model.ProcessInstance;
+import org.activiti.runtime.api.model.builders.ProcessPayloadBuilder;
+import org.activiti.runtime.api.model.impl.APIProcessDefinitionConverter;
+import org.activiti.runtime.api.model.impl.APIProcessInstanceConverter;
+import org.activiti.runtime.api.model.impl.APIVariableInstanceConverter;
+import org.activiti.runtime.api.model.impl.ProcessInstanceImpl;
+import org.activiti.runtime.api.model.payloads.DeleteProcessPayload;
+import org.activiti.runtime.api.model.payloads.GetProcessDefinitionsPayload;
+import org.activiti.runtime.api.model.payloads.GetProcessInstancesPayload;
+import org.activiti.runtime.api.model.payloads.StartProcessPayload;
+import org.activiti.runtime.api.query.Page;
+import org.activiti.runtime.api.query.Pageable;
+import org.activiti.runtime.api.query.impl.PageImpl;
+import org.springframework.security.access.prepost.PreAuthorize;
+
+import java.util.List;
+
+@PreAuthorize("hasRole('ACTIVITI_ADMIN')")
+public class ProcessAdminRuntimeImpl implements ProcessAdminRuntime {
+
+    private final RepositoryService repositoryService;
+
+    private final APIProcessDefinitionConverter processDefinitionConverter;
+
+    private final RuntimeService runtimeService;
+
+    private final APIProcessInstanceConverter processInstanceConverter;
+
+    private final APIVariableInstanceConverter variableInstanceConverter;
+
+    private final ProcessRuntimeConfiguration configuration;
+
+    public ProcessAdminRuntimeImpl(RepositoryService repositoryService,
+                                   APIProcessDefinitionConverter processDefinitionConverter,
+                                   RuntimeService runtimeService,
+                                   APIProcessInstanceConverter processInstanceConverter,
+                                   APIVariableInstanceConverter variableInstanceConverter,
+                                   ProcessRuntimeConfiguration configuration) {
+        this.repositoryService = repositoryService;
+        this.processDefinitionConverter = processDefinitionConverter;
+        this.runtimeService = runtimeService;
+        this.processInstanceConverter = processInstanceConverter;
+        this.variableInstanceConverter = variableInstanceConverter;
+        this.configuration = configuration;
+    }
+
+
+    @Override
+    public ProcessDefinition processDefinition(String processDefinitionId) {
+        org.activiti.engine.repository.ProcessDefinition processDefinition = null;
+        try {
+            processDefinition = repositoryService.getProcessDefinition(processDefinitionId);
+        } catch (Exception internalEx) {
+            // no action, by ID didn't worked, we will try by Key
```

Comment:
@erdemedeiros I am not convinced on having any other parameter.. having 2 already is a nightmare I don't want to introduce a third one. We should always check by key first and then check by ID just in case. Does that sounds like a good compromise and default behaviour? (can you modify that check to do that in that order?)

Reply:
@Salaboy you forgot to tell you thoughts about this comment.

Reply:
@Salaboy looking for the key first seems to be a good compromise. But I think you can keep it  as it is for now.